objects/src/builder/mod.rs

```diff
@@ -153,13 +159,6 @@ impl<T: Rng> NonFungibleAssetBuilder<T> {
         Ok(Self { details_builder })
     }
 
-    pub fn with_details(
-        &mut self,
-        details: &NonFungibleAssetDetails,
-    ) -> Result<NonFungibleAsset, AssetError> {
-        NonFungibleAsset::new(&details)
-    }
```

Comment:
It doesn't make a lot of sense to have this constructor, if the user has that object they can call `NonFungibleAsset::new` directly. This constructor would also make the use of `rng` totally unecessary here.